go.mod

```diff
@@ -28,16 +28,16 @@ require (
 
 require (
 	github.com/golang/protobuf v1.5.3 // indirect
-	golang.org/x/net v0.20.0 // indirect
-	golang.org/x/sys v0.17.0 // indirect
```

Comment:
Don't change these, leave these as they are the replace below will force the later version. Those we can update without the system touching them. The indirect ones are managed by the go module system.

Reply:
Done. Thanks for the feedback!

Reply:
Hi @targetoee, 
We don't manually change indirect dependency.

However, `golang.org/x/sys v0.18.0 // indirect` could be kept since it was resovled by 'go mod tidy' after upgrade to golang.org/x/net v0.23.0.

Here is our previous discussion for indirect/replacement
- https://github.com/apache/yunikorn-k8shim/pull/794#pullrequestreview-1896062777

Reply:
> Don't change these, leave these as they are the replace below will force the later version.

Hi @wilfred-s, I still think we have a valid reason(Fix CVE) to update the automatically generated dependencies that were updated by 'go mod tidy'.

Reply:
I think it's reasonable even though the dependencies will propagate to downstream packages.

Reply:
Except it doesn't work like that. The purpose of 'go mod tidy' is to remove redundant information and 'tidy up' dependencies. We should never be committing any go.mod / go.sum changes which go against what 'go mod tidy' produces, otherwise every other user will have to undo work that it performs every time. Those depended are indirect, and therefore not our responsibility. Instead we use replacements to ensure that our own builds are forced to use newer versions. Please do as Wilfred asks and leave these alone.

Reply:
Let me explain my thoughts step by step,

We changed ‘replace' directive to 'golang.org/x/net v0.23.0' first.
Then run 'go mod tidy' and it will automatilcally update 'golang.org/x/sys' to v0.18.0. (**The minimum required version**)
Below is the tidy result:
<img width="993" alt="image" src="https://github.com/apache/yunikorn-scheduler-interface/assets/26764036/f99494e8-78c1-4dee-8bc9-bb9d24fbf1af">


Since 'replace' directive take higher precedence than indirect dependencies, we should also change the 'replace' directive to 'golang.org/x/sys v0.18.0'. (Otherwise Go will use the outdated version)
So the correct change for this PR is:

<img width="1032" alt="image" src="https://github.com/apache/yunikorn-scheduler-interface/assets/26764036/d28501b8-98b3-4c59-9a87-70e0de7c1c93">

Reply:
Ok, I think I understand now. 'Go mod tidy' updated the indirect version so you are updating the replacement to match. I think that's fine.